File: backend/app/schemas/repository_chat.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ChatMessageRequest(BaseModel):
    message: str


class ChatMessageResponse(BaseModel):
    answer: str
    cited_files: list[str]


class ChatHistoryItem(BaseModel):
    id: int
    role: str
    content: str
    cited_files: list[str]
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/repositoryChat.ts
export interface ChatMessageRequest {
  message: string
}

export interface ChatMessageResponse {
  answer: string
  cited_files: string[]
}

export interface ChatHistoryItem {
  id: number
  role: 'user' | 'assistant'
  content: string
  cited_files: string[]
  created_at: string
}
